### PR TITLE
[MIG] privacy_consent: Migrate to v12

### PR DESCRIPTION
**Pull request by yajo**
- Partner's `opt_out` no longer exists. Using `mail.blacklist` now.
- Tests updated to support that change.
- Test workarounds removed.
- Duplicated-field-name-in-model warning removed.

@Tecnativa

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please fix little flake8 error:
```
privacy_consent/tests/test_consent.py:4:1: F401 'contextlib.contextmanager' imported but unused
```

**Comment by yajo** [comment]
Fixed.

File: privacy_consent/models/mail_template.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from odoo.exceptions import ValidationError
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    @api.constrains("body_html", "model")
+    def _check_consent_links_in_body_html(self):
+        """Body for ``privacy.consent`` templates needs placeholder links."""
+        links = [u"//a[@href='/privacy/consent/{}/']".format(action)
```
Comment (Tardo) [inline privacy_consent/models/mail_template.py]: Python 3 works with unicode strings by default...
https://www.python.org/dev/peps/pep-3120/ says...
```With Python 3, all strings will be Unicode strings, so the original encoding of the source will have no impact at run-time.```
Another related PEP: https://www.python.org/dev/peps/pep-0414/
```

File: privacy_consent/models/privacy_activity.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        # Skip activitys where consent is not required
+        for one in self.with_context(active_test=False) \
+                .filtered("consent_required"):
+            domain = [
+                ("id", "not in", one.mapped("consent_ids.partner_id").ids),
+                ("email", "!=", False),
+            ] + safe_eval(one.subject_domain)
+            # Create missing consent requests
+            for missing in self.env["res.partner"].search(domain):
+                consents |= consents.create({
```
Comment (Tardo) [inline privacy_consent/models/privacy_activity.py]: Here perhaps can use create (multi style)

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
Thanks @Tardo, all  fixed.

File: privacy_consent/models/privacy_consent.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        super_ = super(PrivacyConsent,
+                       self.with_context(mail_create_nolog=True))
+        results = super_.create(vals_list)
+        # Sync the default acceptance status
+        results.sudo()._run_action()
+        return results
+
+    def write(self, vals):
+        """Run server action on update."""
+        result = super(PrivacyConsent, self).write(vals)
```
Comment (Tardo) [inline privacy_consent/models/privacy_consent.py]:
```suggestion
        result = super().write(vals)
```

File: privacy_consent/models/privacy_consent.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        self._run_action()
+        return result
+
+    def message_get_suggested_recipients(self):
+        result = super(PrivacyConsent, self) \
```
Comment (Tardo) [inline privacy_consent/models/privacy_consent.py]:
```suggestion
        result = super() \
```

**Review by Tardo — CHANGES_REQUESTED** [review]

**Review by Tardo — APPROVED** [review]
:+1: code review

**Pull request merged**